### DISCUSSION TITLE
[BUG] Add-PnPFieldToContentType throws error 'Column 'XXX' does not exist. It may have been deleted by another user.'

### DISCUSSION THREAD

**Discussion by HWielenDHS** [pnp/powershell#4659] [General]
I have created different site columns that have to be added to contenttypes in the document libraries of existing SharePoint subsites.

I'm trying to use a PnP PowerShell script for this with the cmdlet 'Add-PnPFieldToContentType'. But whatever I try I always get the error message: 'Column 'XXX' does not exist. It may have been deleted by another user.'

I'm using a Windows 11 PC with PowerShell 7.4.6 en tried different versions of the PnP.PowerShell module up to the most recent version '2.99.90 nightly'.

In my script I first check if the contenttype is available using an 'If'-statement with: `$contentType = Get-PnPContentType -List $libraryName | Where-Object { $_.Name -eq $contentTypeName }`

Secondly I check if the column is available in the document library using using an 'If'-statement with: `$column = Get-PnPField -List $libraryName | Where-Object { $_.InternalName -eq $columnName }`

When everything checks out I use the following code to add the column to the contenttype: `Add-PnPFieldToContentType -Field $column.InternalName -ContentType $contentType.Name`

I have different types of columns: Number, Line of text and Taxonomy but for all of them I get the same error.
For the '-Field' parameter of the Add-PnPFieldToContentType cmdlet I used the properties: 'Title', 'InternalName', 'Id' and the object itself but it makes no difference.

I can add these columns to the contenttype manually in the SharePoint document libraries without any problems.

What am I missing?

**Comment by waaromikniet** [comment]
Does it work when you try your script on a library in the rootsite and not a subsite?

**Reply by HWielenDHS** [reply]
I have got it working now but only when I use the column IDs en contenttype names, like this:

$columnIds = @("a7cb63e1-b47f-434d-bef2-13ffaa625f68", "c1437268-39a4-4ca5-aab4-034df9aaef4b", "2dea25a0-2e4b-4266-899f-9a672d746458")

foreach ($columnId in $columnIds) {
     $column = Get-PnPField -List $library | Where-Object { $_.Id -eq $columnId }

     if ($column -ne $null) {
           Add-PnPFieldToContentType -Field $column -ContentType $contentType.Name
     } else {
           Write-Host "Veld 'Documenttype' not found"
     }
}

**Comment by jackpoz** [comment]
Instead of using
```powershell
Add-PnPFieldToContentType -Field $column.InternalName -ContentType $contentType.Name
```
have you tried using
```powershell
Add-PnPFieldToContentType -Field $column -ContentType $contentType
```
?

That way PnP won't have to find the column and the content type by name again but use the ones found in the previous commands
